Flakey unit-test job in CI failing on TestNonceFromLRU

I believe this is due to adding test.parallel in some places, because this wasn't failing before:

```
--- FAIL: TestNonceFromLRU (0.02s)
{"Timestamp":1556579600711799809,"Time":"2019-04-29T23:13:20Z","Type":"app.log","Logger":"autograph","Hostname":"b02acbc07f9e","EnvVersion":"2.0","Pid":9685,"Severity":4,"Fields":{"msg":"Error sending hawk.validated: write udp 127.0.0.1:40889-\u003e127.0.0.1:8125: write: connection refused"}}
    authorize_test.go:150: Second nonce "Rtt2aEhJ" not found in cache, should have been present
    authorize_test.go:151: nonces: [eJlRk/4P]
```

--- a/authorize_test.go
+++ b/authorize_test.go
@@ -132,6 +132,9 @@ func TestNonceFromLRU(t *testing.T) {
 		0)
 	req.Header.Set("Authorization", auth1.RequestHeader())
 	_, _, err = ag.authorizeHeader(req)
+	if err != nil {
+		t.Fatalf("error authorizing header for first request: %s", err)
+	}
 
 	auth2 := hawk.NewRequestAuth(req,
 		&hawk.Credentials{
@@ -141,6 +144,9 @@ func TestNonceFromLRU(t *testing.T) {
 		0)
 	req.Header.Set("Authorization", auth2.RequestHeader())
 	_, _, err = ag.authorizeHeader(req)
+	if err != nil {
+		t.Fatalf("error authorizing header for second request: %s", err)
+	}
 
 	if ag.nonces.Contains(auth1.Nonce) {
 		t.Errorf("First nonce %q found in cache, should have been removed", auth1.Nonce)

--- a/authorize_test.go
+++ b/authorize_test.go
@@ -117,8 +117,6 @@ func TestDuplicateNonce(t *testing.T) {
 }
 
 func TestNonceFromLRU(t *testing.T) {
-	t.Parallel()
-
 	req, err := http.NewRequest("POST", "http://foo.bar/sign/data", nil)
 	if err != nil {
 		t.Fatal(err)

--- a/tools/autograph-monitor/contentsignature_test.go
+++ b/tools/autograph-monitor/contentsignature_test.go
@@ -6,10 +6,12 @@ import (
 	"fmt"
 	"io/ioutil"
 	"log"
+	"net"
 	"net/http"
 	"os"
 	"strings"
 	"testing"
+	"time"
 
 	"go.mozilla.org/autograph/signer/contentsignaturepki"
 )
@@ -25,7 +27,13 @@ func TestVerifyContentSignature(t *testing.T) {
 		})
 		log.Fatal(http.ListenAndServe(":64320", nil))
 	}()
-	err := verifyContentSignature(ValidMonitoringContentSignature)
+	setupTimeout, _ := time.ParseDuration("10s")
+	conn, err := net.DialTimeout("tcp", net.JoinHostPort("", "64320"), setupTimeout)
+	if conn != nil {
+		conn.Close()
+	}
+
+	err = verifyContentSignature(ValidMonitoringContentSignature)
 	if err != nil {
 		t.Fatalf("Failed to verify monitoring content signature: %v", err)
 	}

